docs/rfcs/env-vars.md

```diff
@@ -0,0 +1,222 @@
+not expect expansion to happen. Three examples where this is unexpected
+are the following:
+
+1.  **Opaque values such as passwords that contain `$`** (issue
+    [#8215](https://github.com/open-telemetry/opentelemetry-collector/issues/8215)).
+    If the $ is followed by an alphanumeric character or one of the
+    special characters, it's going to lead to false positives.
+2.  **Prometheus relabel config** (issue
+    [`contrib#9984`](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/9984)).
+    Prometheus uses `${1}` in some of its configuration values. We
+    resolve this to the value of the environment variable with name
+    '`1`'.
+3.  **Other uses of $** (issue
+    [`contrib#11846`](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/11846)).
+    If a product requires the use of `$` in some field, we would most
+    likely interpret it as an environment variable. This is not
+    intuitive for users.
+
+#### Unexpected type casting
+
+When using the env syntax we parse its value as YAML. Even if you are
+familiar with YAML, because of the implicit type casting rules and the
+way we store intermediate values, we can get unintuitive results.
+
+The most clear example of this is issue
+[*#8565*](https://github.com/open-telemetry/opentelemetry-collector/issues/8565):
+When setting a variable to value `0123` and using it in a string-typed
+field, it will end up as the string `"83"` (where as the user would
+expect the string to be `0123`).
+
+#### We are less restrictive than the Configuration WG
+
+The Configuration WG defines an [*environment variable expansion feature
+for SDK
+configurations*](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/file-configuration.md#environment-variable-substitution).
+This accepts only non empty alphanumeric + underscore identifiers
+starting with alphabetic or underscore. If the Configuration WG were to
+expand this in the future (e.g. to include other features present in
+Bash-like syntax as in [opentelemetry-specification/pull/3948](https://github.com/open-telemetry/opentelemetry-specification/pull/3948)), we would not be able to expand our braces syntax to
+support new features without breaking users.
+
+## Desired behavior
+
+*This section is written as if the changes were already implemented.*
+
+The Collector supports **two** different syntaxes for environment
+variable resolution:
+
+1.  The *braces syntax*, `${ENV}`.
+2.  The *env provider syntax*, `${env:ENV}`.
+
+These both have **the same character set and behavior**. They both use
+the env provider under the hood. This means we support the exact same
+syntax as the Configuration WG.
+
+The naked syntax supported in BASH is not supported in the Collector.
+Escaping is supported by using two dollar signs. Escaping is also
+honored for unsupported identifiers like `${1}` (i.e. anything that 
```

Comment:
Based on our discussion in [this thread](https://github.com/open-telemetry/opentelemetry-collector/pull/9854#discussion_r1549442566) this sentence should be updated to clarify that `${1}` is no supported.